tests/Unit/Entity/BannerSelection/CampaignCollectionTest.php

```diff
@@ -4,30 +4,46 @@
 
 namespace WMDE\BannerServer\Tests\Unit\Entity\BannerSelection;
 
+use WMDE\BannerServer\Entity\BannerSelection\Banner;
+use WMDE\BannerServer\Entity\BannerSelection\Bucket;
 use WMDE\BannerServer\Entity\BannerSelection\Campaign;
 use WMDE\BannerServer\Entity\BannerSelection\CampaignCollection;
+use WMDE\BannerServer\Tests\Utils\FakeRandomInteger;
 
 class CampaignCollectionTest extends \PHPUnit\Framework\TestCase {
 
+	private function getTestbucket(): Bucket {
+		return new Bucket(
+			'test',
+			new Banner( 'TestMain' )
+		);
+	}
+
 	public function test_given_date_returns_active_campaign() {
 		$campaignCollection = new CampaignCollection(
 			new Campaign(
 				'C18_WMDE_Test_future',
 				new \DateTime( '2099-10-01 14:00:00' ),
 				new \DateTime( '2099-10-31 14:00:00' ),
-				[]
+				1,
```

Comment:
Please use a constant for the impression count to avoid having to jump to the constructor params for reference.